### PR TITLE
Underground Zombie Landmarks

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Adds Zombie Landmarks that makes it easier to spawn in zombies for Dchat without them just... standing around. Comes in three variants, one that allows one zombie to spawn at it before it deletes itself, one that does the same but three times, and one that does it infinitely.

## Why It's Good For The Game

Last zombie event it kinda suffered because staff had to spawn zombies in bushes to hide them, so this'll hide them by making the spawnpoints invisible to everyone but observers.

## Changelog

:cl:
admin: Added zombie spawnpoints for events.
/:cl:

### CODE REVIEW

**Comment by fira** [comment]
why the invis shenanigans?

**Comment by Geevies** [comment]
> why the invis shenanigans?

it keeps regular landmarks hidden, and lets observers see the zombie landmarks dotted around the map. as for why it's set in /New, i have no idea and i think it's out of scope to change it here. i assume it's done for a reason

File: code/game/objects/effects/landmarks/landmarks.dm [hunk 1]
```diff
@@ -405,3 +407,47 @@
...
+
+/obj/effect/landmark/zombie/Destroy()
+	GLOB.zombie_landmarks -= src
+	return ..()
+
+/obj/effect/landmark/zombie/proc/spawn_zombie(var/mob/dead/observer/observer)
+	if(!infinite_spawns)
+		spawns_left--
+	if(spawns_left <= 0)
+		GLOB.zombie_landmarks -= src
```
Comment (stanalbatross) [inline code/game/objects/effects/landmarks/landmarks.dm:433]: not qdel?

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/landmarks/landmarks.dm (hunk 1 above)
Comment (fira) [reply code/game/objects/effects/landmarks/landmarks.dm:433]: yeah you'd want it deleted i think

**Review by fira — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/landmarks/landmarks.dm [hunk 2]
```diff
@@ -405,3 +407,47 @@
...
+
+/obj/effect/landmark/zombie/proc/handle_zombie_spawn(var/mob/dead/observer/observer)
+	var/mob/living/carbon/human/zombie = new /mob/living/carbon/human(loc)
+	if(!zombie.hud_used)
+		zombie.create_hud()
+	arm_equipment(zombie, /datum/equipment_preset/other/zombie, randomise = TRUE, count_participant = TRUE, mob_client = observer.client, show_job_gear = TRUE)
+	observer.client.eye = zombie
+	observer.mind.transfer_to(zombie)
+	if(spawns_left <= 0)
+		qdel(src)
```
Comment (Geevies) [inline code/game/objects/effects/landmarks/landmarks.dm:447]: qdel is handled here @stanalbatross @fira

**Review by fira — APPROVED** [review]

**Review by FransFeiffer — APPROVED** [review]
Sprites are fine :D

**Pull request merged**